lms/views/admin.py

```diff
@@ -71,16 +73,14 @@ def update_instance(self):
             "sections_enabled" in self.request.params
             and self.request.params["sections_enabled"] == "on"
         )
+        ai.settings.set("canvas", "sections_enabled", sections_enabled)
+
         groups_enabled = (
             "groups_enabled" in self.request.params
             and self.request.params["groups_enabled"] == "on"
         )
+        ai.settings.set("canvas", "groups_enabled", groups_enabled)
 
-        self.application_instance_service.update_settings(
-            ai,
-            canvas_sections_enabled=sections_enabled,
-            canvas_groups_enabled=groups_enabled,
-        )
```

Comment:
I assume any interesting gate keeping that was happening in `update_settings()` can equally well happen in `set()`?

Reply:
Yeah, but I don't think we need the gate keeping